command/resource/client/grpc-config.go

```diff
@@ -4,40 +4,134 @@
 package client
 
 import (
+	"fmt"
 	"os"
+	"strconv"
+	"strings"
 )
 
 const (
 	// GRPCAddrEnvName defines an environment variable name which sets the gRPC
 	// server address for the consul CLI.
 	GRPCAddrEnvName = "CONSUL_GRPC_ADDR"
+
+	// GRPCTLSEnvName defines an environment variable name which sets the gRPC
+	// communication mode. Default is false in plaintext mode.
+	GRPCTLSEnvName = "CONSUL_GRPC_TLS"
+
+	// GRPCTLSVerifyEnvName defines an environment variable name which sets
+	// whether to disable certificate checking.
+	GRPCTLSVerifyEnvName = "CONSUL_GRPC_TLS_VERIFY"
+
+	// GRPCClientCertEnvName defines an environment variable name which sets the
+	// client cert file to use for talking to Consul over TLS.
+	GRPCClientCertEnvName = "CONSUL_GRPC_CLIENT_CERT"
+
+	// GRPCClientKeyEnvName defines an environment variable name which sets the
+	// client key file to use for talking to Consul over TLS.
+	GRPCClientKeyEnvName = "CONSUL_GRPC_CLIENT_KEY"
+
+	// GRPCCAFileEnvName defines an environment variable name which sets the
+	// CA file to use for talking to Consul gRPC over TLS.
+	GRPCCAFileEnvName = "CONSUL_GRPC_CACERT"
+
+	// GRPCCAPathEnvName defines an environment variable name which sets the
+	// path to a directory of CA certs to use for talking to Consul gRPC over TLS.
+	GRPCCAPathEnvName = "CONSUL_GRPC_CAPATH"
 )
 
 type GRPCConfig struct {
+	// Address is the optional address of the Consul server in format of host:port.
+	// It doesn't include schema
 	Address string
+
+	// GRPCTLS is the optional boolean flag to determine the communication protocol
+	GRPCTLS bool
+
+	// GRPCTLSVerify is the optional boolean flag to disable certificate checking.
+	// Set to false only if you want to skip server verification
+	GRPCTLSVerify bool
+
+	// CertFile is the optional path to the certificate for Consul
+	// communication. If this is set then you need to also set KeyFile.
+	CertFile string
+
+	// KeyFile is the optional path to the private key for Consul communication.
+	// If this is set then you need to also set CertFile.
+	KeyFile string
+
+	// CAFile is the optional path to the CA certificate used for Consul
+	// communication, defaults to the system bundle if not specified.
+	CAFile string
+
+	// CAPath is the optional path to a directory of CA certificates to use for
+	// Consul communication, defaults to the system bundle if not specified.
+	CAPath string
 }
 
 func GetDefaultGRPCConfig() *GRPCConfig {
 	return &GRPCConfig{
-		Address: "localhost:8502",
+		Address:       "127.0.0.1:8502",
+		GRPCTLSVerify: true,
```

Comment:
Is tls the default now? That mean, to use the cli you will need to setup the certs, right?

Reply:
Thank you! This is a great catch! That was my mistake, it's supposed to be `plaintext` mode, just changed it.